docs/table.md

```diff
@@ -0,0 +1,377 @@
+      },
+      // Set the style for cells based on their row
+      rowStyles: (i) => {
+        if (i === 0) return { border: { top: 2 }, borderColor: { top: "black" } };
+        if (i === 3) return { border: { bottom: 2 }, borderColor: { bottom: "black" } };
+      },
+      data: [
+        ["Header 1", "Header 2", "Header 3"],
+        ["Sample value 1", "Sample value 2", "Sample value 3"],
+        ["Sample value 1", "Sample value 2", "Sample value 3"],
+        ["Sample value 1", "Sample value 2", "Sample value 3"],
+      ],
+    })
+
+![10]()
+
+---
+
+Zebra style
+
+    doc.table({
+      rowStyles: (i) => {
+        if (i % 2 === 0) return { backgroundColor: "#ccc" };
+      },
+      data: [
+        ["Sample value 1", "Sample value 2", "Sample value 3"],
+        ["Sample value 1", "Sample value 2", "Sample value 3"],
+        ["Sample value 1", "Sample value 2", "Sample value 3"],
+        ["Sample value 1", "Sample value 2", "Sample value 3"],
+        ["Sample value 1", "Sample value 2", "Sample value 3"],
+      ],
+    })
+
+![11]()
+
+---
+
+### Optional border
+
+    doc.table({
+      data: [
+        [
+          { border: [true, false, false, false], backgroundColor: "#eee", content: "border:\n[true, false, false, false]" },
+          { border: false, backgroundColor: "#ddd", content: "border:\nfalse" },
+          { border: true, backgroundColor: "#eee", content: "border:\ntrue" },
+        ],
+        [
+          { rowSpan: 3, border: true, backgroundColor: "#eef", content: "rowSpan: 3\n\nborder:\ntrue" },
+          { border: undefined, backgroundColor: "#eee", content: "border:\nundefined (default)" },
+          { border: [false, false, false, true], backgroundColor: "#ddd", content: "border:\n[false, false, false, true]" },
+        ],
+        [
+          { colSpan: 2, border: true, backgroundColor: "#efe", content: "colSpan: 2\n\nborder:\ntrue" },
+        ],
+        [
+          { border: 0, backgroundColor: "#eee", content: "border:\n0 (same as false)" },
+          { border: [false, true, true, false], backgroundColor: "#ddd", content: "border:\n[false, true, true, false]" },
+        ],
+      ],
+    })
+
+![12]()
+
+---
+
+    doc.table({
+      defaultStyle: { border: false, width: 60 },
+      data: [
+        ["", "column 1", "column 2", "column 3"],
+        [
+          "row 1",
+          {
+            rowSpan: 3,
+            colSpan: 3,
+            border: true,
+            backgroundColor: "#ccc",
+            content:
+              "rowSpan: 3\ncolSpan: 3\n\nborder:\n[true, true, true, true]",
+          },
+        ],
+        ["row 2"],
+        ["row 3"],
+      ],
+    })
+
+![13]()
+
+---
+
+When defining multiple styles, the cells follow the precedence:
+
+1. `defaultStyle`
+2. `columnStyles`
+3. `rowStyles`
+4. `cellStyle`
+
+so if a table was:
+
+    doc.table({
+      defaultStyle: { border: 1 },
```

Comment:
```suggestion
      defaultStyles: { border: 1 },
```

would be better `defaultStyles`?

Reply:
since there can only be one default it is always singular (unlike the columnStyles and rowStyles)

Reply:
@blikblum are you still wanting this renamed, or are you happy with my explanation?

Reply:
I've seem both API designs with singular and plural names of a set of values `style`, `styles`, `default`, `defaults`. 

Lets go with web pataform that uses singular for "style" attribute